move/sources/canvas/token.move

```diff
@@ -382,10 +383,13 @@ module addr::canvas_token {
             let now = now_seconds();
             if (smart_table::contains(&canvas_.last_contribution_s, caller_addr)) {
```

Comment:
it would make sense to move a lot of this `draw once` logic into a private function, and have an internal `draw_multiple`- as right now if I submit 50 pixels, we're calling all of the checks many times, even though we know they would be okay after the first time